### ISSUE TITLE
Unbundle youtube_dl

### ISSUE THREAD

**Issue by benjaoming** [learningequality/ka-lite#3696]
The library is only used for videodownload, but is the exact usage for users or dev?

```
➜  ka-lite git:(windows-kalite-command) ack-grep --python youtube_dl kalite
kalite/updates/management/commands/videodownload.py
4:import youtube_dl
8:from youtube_dl.utils import DownloadError
33:    yt_dl = youtube_dl.YoutubeDL({'outtmpl': video_file_download_path, "quiet": quiet})
168:                        def youtube_dl_cb(stats, progress_callback, *args, **kwargs):
176:                        scrape_video(video.youtube_id, quiet=not settings.DEBUG, callback=partial(youtube_dl_cb, progress_callback=progress_callback))

kalite/basetests/tests.py
111:        # attribute found is a python module like `youtube_dl.version`, so we recurse
231:        "youtube_dl": "2014.12.10.3",
```

**Comment by rtibbles** [comment]
Care to elaborate?

**Comment by benjaoming** [comment]
I don't know what happened, I was trying to type in a description and when I tabbed back to then window, it was posted :)

**Comment by benjaoming** [comment]
There, description added :)

**Comment by benjaoming** [comment]
This package is important to keep updated, right? Youtube changes, so we should at least keep it in a requirements file and maybe not even pin the version number?

**Comment by benjaoming** [comment]
Jamie:

> still being used by end users
> for downloading dubbed videos in some languages

**Comment by benjaoming** [comment]
@jamalex is it still true that end-users are using youtube_dl ?

It's a big chunk of code to get rid of, and since we are re-encoding lots of videos, it'd be cool to get rid of :)

**Comment by rtibbles** [comment]
Looks like this was fixed in #4688, @benjaoming please reopen if not true.

**Comment by benjaoming** [comment]
It's fixed and youtube_dl is running in the newest version at the time of the PR.
